plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/GitHook.kt

```diff
@@ -54,13 +54,12 @@ private fun postCheck(
             git add ${'$'}file
         fi
     done
-    """.trimIndent()
```

Comment:
Doesn't some kind of trimming need to happen here?

Reply:
@JLLeitschuh No because if we keep it, `formatUsesGradleExitCode` fails to find `gradle_command_exit_code=$?` as there is indentation before :

<details><summary>With trimming (original)</summary>

```
Expecting:
 <["#!/bin/sh",
 "######## KTLINT-GRADLE HOOK START ########",
 "",
 "    CHANGED_FILES="$(git --no-pager diff --name-status --no-color --cached | awk '$1 != "D" && $NF ~ /\.kts?$/ { print $NF }')"",
 "",
 "    if [ -z "$CHANGED_FILES" ]; then",
 "        echo "No Kotlin staged files."",
 "        exit 0",
 "    fi;",
 "",
 "    echo "Running ktlint over these files:"",
 "    echo "$CHANGED_FILES"",
 "",
 "    diff=.git/unstaged-ktlint-git-hook.diff",
 "    git diff --color=never > $diff",
 "    if [ -s $diff ]; then",
 "      git apply -R $diff",
 "    fi",
 "",
 "    ./gradlew --quiet ktlintFormat -PinternalKtlintGitFilter="$CHANGED_FILES"",
 "    gradle_command_exit_code=$?",
 "",
 "    echo "Completed ktlint run."",
 "    echo "$CHANGED_FILES" | while read -r file; do",
 "    if [ -f $file ]; then",
 "        git add $file",
 "    fi",
 "done",
 "",
 "    if [ -s $diff ]; then",
 "      git apply --ignore-whitespace $diff",
 "    fi",
 "    rm $diff",
 "    unset diff",
 "",
 "    echo "Completed ktlint hook."",
 "    exit $gradle_command_exit_code",
 "######## KTLINT-GRADLE HOOK END ########"]>
to contain:
 <["gradle_command_exit_code=$?"]>
but could not find:
 <["gradle_command_exit_code=$?"]>
```
</details>

<details><summary>Without trimming</summary>

```
"######## KTLINT-GRADLE HOOK START ########",
"",
"CHANGED_FILES="$(git --no-pager diff --name-status --no-color --cached | awk '$1 != "D" && $NF ~ /\.kts?$/ { print $NF }')"",
"",
"if [ -z "$CHANGED_FILES" ]; then",
"    echo "No Kotlin staged files."",
"    exit 0",
"fi;",
"",
"echo "Running ktlint over these files:"",
"echo "$CHANGED_FILES"",
"",
"diff=.git/unstaged-ktlint-git-hook.diff",
"git diff --color=never > $diff",
"if [ -s $diff ]; then",
"  git apply -R $diff",
"fi",
"",
"./gradlew --quiet ktlintFormat -PinternalKtlintGitFilter="$CHANGED_FILES"",
"gradle_command_exit_code=$?",
"",
"echo "Completed ktlint run."",
"",
"echo "$CHANGED_FILES" | while read -r file; do",
"    if [ -f $file ]; then",
"        git add $file",
"    fi",
"done",
"",
"",
"if [ -s $diff ]; then",
"  git apply --ignore-whitespace $diff",
"fi",
"rm $diff",
"unset diff",
"",
"echo "Completed ktlint hook."",
"exit $gradle_command_exit_code",
"######## KTLINT-GRADLE HOOK END ########"]>
```
</details>

But if you tell me that the first indentation with `trimIndent` (while it's supposed to remove indentation) is expected, then I'll add it back and look for `exit $gradle_command_exit_code` with indentation in the test instead.